sdks/python/apache_beam/runners/dataflow/internal/apiclient.py

```diff
@@ -923,8 +923,9 @@ def job_id_for_name(self, job_name):
           pageToken=token)
       response = self._client.projects_locations_jobs.List(request)
       for job in response.jobs:
-        if (job.name == job_name and job.currentState ==
-            dataflow.Job.CurrentStateValueValuesEnum.JOB_STATE_RUNNING):
+        if (job.name == job_name and job.currentState in
+            [dataflow.Job.CurrentStateValueValuesEnum.JOB_STATE_RUNNING,
+             dataflow.Job.CurrentStateValueValuesEnum.JOB_STATE_DRAINING]):
```

Comment:
Add a test to apiclient_test so that this behavior is not reverted in the future?

Reply:
I created this change with the github single-file in-browser editor, and now I don't think I can amend it with another file change. Can someone with the infrastructure already set up take this over?

Reply:
Let's get this change in, in that case. We can add tests after. Github's UI is limited to single file edits.